### PR TITLE
Adding downloads of 'arm' and 'arm64'

### PR DESCRIPTION
**Pull request by ghost**
Adding downloads of 'arm' and 'arm64'

Addresses: https://github.com/dotnet/cli/issues/10302

### CODE REVIEW

File: scripts/obtain/dotnet-install.ps1 [hunk 1]
```diff
@@ -151,18 +151,19 @@ function Invoke-With-Retry([ScriptBlock]$ScriptBlock, [int]$MaxAttempts = 3, [in
 function Get-Machine-Architecture() {
     Say-Invocation $MyInvocation
 
-    # possible values: AMD64, IA64, x86
```
Comment (peterhuene) [inline scripts/obtain/dotnet-install.ps1]: What, no `IA64` anymore?! :trollface:

**Review by peterhuene — APPROVED** [review]

File: scripts/obtain/dotnet-install.ps1 [hunk 2]
```diff
@@ -151,18 +151,19 @@ function Invoke-With-Retry([ScriptBlock]$ScriptBlock, [int]$MaxAttempts = 3, [in
...
 
-# TODO: Architecture and CLIArchitecture should be unified
 function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
     Say-Invocation $MyInvocation
 
     switch ($Architecture.ToLower()) {
         { $_ -eq "<auto>" } { return Get-CLIArchitecture-From-Architecture $(Get-Machine-Architecture) }
-        { ($_ -eq "amd64") -or ($_ -eq "x64") } { return "x64" }
-        { $_ -eq "x86" } { return "x86" }
+        { ($_ -like "amd64") -or ($_ -like "x64") } { return "x64" }
```
Comment (peterhuene) [inline scripts/obtain/dotnet-install.ps1]: One question from powershell-ignorant me: why `like` vs `eq`?

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.ps1]: "-like" is case-insensitive
It for when someone types "X86" instead of "x86" on the command-line

Comment (peterhuene) [reply scripts/obtain/dotnet-install.ps1]: Is it a whole-string match or a sub-string match?  I assume `eq` was whole.

Comment (peterhuene) [reply scripts/obtain/dotnet-install.ps1]: Erm, never mind.  I think I was looking at something else.  This isn't regex, just some wild character characters.  Assuming this isn't substring matching, 👍

Comment (johnbeisner) [reply scripts/obtain/dotnet-install.ps1]: "-eq" is already case-insensitive in Powershell; my mistake

**Review by mikem8361 — APPROVED** [review]

**Comment by livarcocc** [comment]
We don't have a win-arm64 SDK available yet.

win-arm binaries are being added in 2.2.1xx and win-arm64 binaries are going to be added in 3.0.

The shell scripts contained it because we have linux-arm and linux-arm64 binaries available since 2.1.

Should we maybe add win-arm now and add win-arm64 once we enable those builds?

**Comment by rido-min** [comment]
Installing Win-ARM64 fails with:

```
 .\dotnet-install.ps1 -Channel master -Architecture arm64
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview-009756/dotnet-sdk-3.0.100-preview-009756-win-arm64.zip
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview-009756/dotnet-sdk-3.0.100-preview-009756-win-arm64.zip
dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview-009756/dotnet-dev-win-arm64.3.0.100-preview-009756.zip
dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview-009756/dotnet-dev-win-arm64.3.0.100-preview-009756.zip
```
I've found this [zip](https://dotnetcli.blob.core.windows.net/dotnet/Runtime/master/dotnet-runtime-latest-win-arm64.zip), but not sure how to install.

**Pull request merged**